### PR TITLE
Schema and model change to add sequence to workflow

### PR DESCRIPTION
**Pull request by bzwei**
Backend support to enable workflow sequencing management per tenant using ActsAsList.

When a new workflow is created, the sequence is automatically assigned to the next available value if it is not provided by the caller.

Workflow update API allows the caller to directly update. No dedicated API and service class are needed.

When a workflow sequence is changed, all affected workflows' sequences will be automatically updated.

### CODE REVIEW

File: app/services/workflow_sequence_service.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class WorkflowSequenceService
+  def list_by_sequence
+    ActsAsTenant.current_tenant.workflows
+  end
+
+  def update_sequences(workflow_ids)
```
Comment (hsong-rh) [inline app/services/workflow_sequence_service.rb]: This `workflow_ids` should contain all existing workflow ids, right? Partial of workflow ids will mess the `sequence`. Suggest to rename to `update_all_sequences`, if this is the case.

**Comment by bzwei** [comment]
I have removed the service class as the design is now much simplified since the initial work.

File: spec/models/workflow_spec.rb [hunk 2]
```diff
@@ -8,6 +8,40 @@
...
+      old_last = Workflow.all.last
+      new_workflow = create(:workflow)
+      expect(new_workflow.sequence).to be > old_last.sequence
+    end
+
+    it 'moves up an workflow sequence' do
+      old_ids = Workflow.all.pluck(:id)
+      Workflow.find(old_ids[3]).update(:sequence => Workflow.find(old_ids[1]).sequence)
+      new_ids = Workflow.all.pluck(:id)
+      expect(new_ids).to eq([old_ids[0], old_ids[3], old_ids[1], old_ids[2], old_ids[4]])
```
Comment (gmcculloug) [inline spec/models/workflow_spec.rb]: Same, drop `new_ids` and just use `Workflow.pluck(:id)` in the `expect` call.

File: spec/models/workflow_spec.rb [hunk 3]
```diff
@@ -8,6 +8,40 @@
...
+    end
+
+    it 'moves down an workflow sequence' do
+      old_ids = Workflow.all.pluck(:id)
+      Workflow.find(old_ids[1]).update(:sequence => Workflow.find(old_ids[3]).sequence)
+      new_ids = Workflow.all.pluck(:id)
+      expect(new_ids).to eq([old_ids[0], old_ids[2], old_ids[3], old_ids[1], old_ids[4]])
```
Comment (gmcculloug) [inline spec/models/workflow_spec.rb]: Same here.

**Pull request merged**